### PR TITLE
Reverse proxy authentication to Woodpeck and Gitea

### PR DESCRIPTION
**Pull request by pboguslawski**
This mod adds option to authenticate user using HTTP header set by
reverse proxy. It forwards specified HTTP header with authenticated
username in requests to Gitea.

Requirements:

* Gitea must be configured for reverse proxy authentication and
  must accept HTTP header auth in API calls (https://github.com/go-gitea/gitea/pull/15119).

* Woodpecker must use its external host address in status URL-s
  sent to Gitea (https://github.com/laszlocph/woodpecker/pull/175).

To enable set the following variables in woodpecker server
environment (woodpecker running behind reverse proxy):

* internal woodpecker server URL, i.e.:

  `WOODPECKER_HOST_INTERNAL=http://192.168.1.100:8000`

* enable reverse proxy auth in woodpecker and forwarding auth
  header to gitea:

  `WOODPECKER_GITEA_REV_PROXY_AUTH=true`

* set name of header with authenticated username (set by
  reverse proxy), i.e.:

  `WOODPECKER_GITEA_REV_PROXY_AUTH_HEADER=X-Forward-Username`

Author-Change-Id: IB#1107569

### CODE REVIEW

File: cmd/drone-server/server.go [hunk 1]
```diff
@@ -513,6 +521,16 @@ var flags = []cli.Flag{
...
 		Usage:  "server-side enforcement policy on the minimum amount of time a client should wait before sending a keepalive ping.",
 	},
+	cli.BoolFlag{
+		EnvVar: "WOODPECKER_GITEA_REV_PROXY_AUTH",
+		Name:   "gitea-rev-proxy-auth",
+		Usage:  "enable gitea authentication using HTTP header specified in WOODPECKER_GITEA_REV_PROXY_AUTH_HEADER",
+	},
+	cli.StringFlag{
+		EnvVar: "WOODPECKER_GITEA_REV_PROXY_AUTH_HEADER",
+		Name:   "gitea-rev-proxy-auth-header",
```
Comment (6543) [inline cmd/drone-server/server.go:531]: wonder if we can make it more generic?

Comment (pboguslawski) [reply cmd/drone-server/server.go:531]: Please propose better name (WOODPECKER_TO_VCS_AUTH_HEADER maybe)?

Comment (6543) [reply cmd/drone-server/server.go:531]: `WOODPECKER_PROXY_AUTH_HEADER` `proxy-auth-header` ?

Comment (pboguslawski) [reply cmd/drone-server/server.go:531]: > WOODPECKER_PROXY_AUTH_HEADER proxy-auth-header ?

Ambiguous - one may think its header from rev proxy to woodpecker (its header sent from woodpecker to vcs).

Comment (6543) [reply cmd/drone-server/server.go:531]: Hmm let me rethink

**Comment by 6543** [comment]
@pboguslawski can you rebase this pull ontop of master?

**Comment by pboguslawski** [comment]
> @pboguslawski can you rebase this pull ontop of master?

Done.

**Comment by 6543** [comment]
ok now **all** gitea pulls are now merged :)

@pboguslawski the heavy woodpecker refactoring is now mostly done - wana rebase?

**Comment by 6543** [comment]
@pboguslawski :eyes:

**Comment by anbraten** [comment]
As Woodpecker is now supporting Gitea login via OAuth which is quite convenient IMO, I would close this for now. Please feel free to reopen if you still want to get this in.

**Comment by pboguslawski** [comment]
OAuth should be optional not mandatory. Think of authenticating gitea and woodpecker user using reverse proxy and http header.

This PR was replaced with 837 (adjusted to current master).

**Pull request closed without merging**